lib/IMAP/MessageMapper.php

```diff
@@ -242,14 +243,14 @@ public function findByIds(Horde_Imap_Client_Base $client,
 
 		/** @var Horde_Imap_Client_Data_Fetch[] $fetchResults */
 		$fetchResults = iterator_to_array($client->fetch($mailbox, $query, [
-			'ids' => new Horde_Imap_Client_Ids($ids),
+			'ids' => $ids,
 		]), false);
 
 		if (empty($fetchResults)) {
 			$this->logger->debug("findByIds in $mailbox got " . count($ids) . " UIDs but found none");
 		} else {
-			$minRequested = $ids[0];
-			$maxRequested = $ids[count($ids) - 1];
+			$minRequested = $ids->ids[0];
+			$maxRequested = $ids->ids[count($ids) - 1];
```

Comment:
Possibly?

```suggestion
			$maxRequested = $ids->count() - 1;
```

Reply:
Let's meet in the middle with `$ids->ids[$ids->count() - 1];`? count-1 would not give me the value of the last element.

Reply:
```
 * @property-read string $max  The largest ID (@since 2.20.0).
 * @property-read string $min  The smallest ID (@since 2.20.0).
 * @property-read string $range_string  Generates a range string consisting of
 *                                      all messages between begin and end of
 *                                      ID list.
```

It seems that Horde_Imap_Client_Ids has max, min and even a range_string out of the box ;)

Reply:
How many engineers does it take to …